pandas/tests/indexing/test_floats.py

```diff
@@ -61,109 +61,115 @@ def test_scalar_error(self):
-                            len(i), len(i)), index=i, columns=i)]:
-
-                # getting
-                for idxr, getitem in [(lambda x: x.ix, False),
-                                      (lambda x: x.iloc, False),
-                                      (lambda x: x, True)]:
+            # getting
+            for idxr, getitem, iloc, loc in [
+                    (lambda x: x.ix, False, False, False),
+                    (lambda x: x.iloc, False, True, False),
+                    (lambda x: x, True, False, False),
+                    (lambda x: x.loc, False, False, True)]:
 
+                if getitem and isinstance(obj, DataFrame):
                     # gettitem on a DataFrame is a KeyError as it is indexing
                     # via labels on the columns
-                    if getitem and isinstance(s, DataFrame):
-                        error = KeyError
-                        msg = r"^3(\.0)?$"
-                    else:
-                        error = TypeError
-                        msg = (r"cannot do (label|index|positional) indexing"
-                               r" on {klass} with these indexers \[3\.0\] of"
-                               r" {kind}|"
-                               "Cannot index by location index with a"
-                               " non-integer key"
-                               .format(klass=type(i), kind=str(float)))
-                    with catch_warnings(record=True):
-                        with pytest.raises(error, match=msg):
-                            idxr(s)[3.0]
-
-                # label based can be a TypeError or KeyError
-                if s.index.inferred_type in ['string', 'unicode', 'mixed']:
+                    error = KeyError
+                    msg = r"^3(\.0)?$"
+                elif loc and obj.index.inferred_type in [
+                        'string', 'unicode']:
+                    # label based can be a KeyError with object Index types
                     error = KeyError
                     msg = r"^3$"
                 else:
                     error = TypeError
-                    msg = (r"cannot do (label|index) indexing"
-                           r" on {klass} with these indexers \[3\.0\] of"
-                           r" {kind}"
-                           .format(klass=type(i), kind=str(float)))
+                    if iloc:
+                        msg = ("Cannot index by location index with a"
+                               " non-integer key")
+                    else:
+                        msg = (r"cannot do (label|index) indexing"
+                               r" on {klass} with these indexers \[3\.0\] of"
+                               r" {kind}"
+                               .format(klass=type(i), kind=str(float)))
                 with pytest.raises(error, match=msg):
-                    s.loc[3.0]
+                    idxr(obj)[3.0]
 
-                # contains
-                assert 3.0 not in s
```

Comment:
i guess that maybe this should have been `assert 3.0 not in i`

Reply:
what is i?